SConstruct

```diff
@@ -0,0 +1,55 @@
+import os
+
+VariantDir('build', 'kernel', duplicate=0)
+env = Environment(
+    ENV={'PATH': os.environ['PATH']},
+    REPO_URL='https://github.com/XyrisOS/xyris',
+    GIT_COMMIT='TODO',
+    VERSION_ID=(1, 0, 0),
+    VERSION_NAME='Pheonix',
+    CFLAGS=[
+        '-nostdlib', '-nodefaultlibs', '-ffreestanding', '-fstack-protector-all',
+        '-fno-builtin', '-fno-omit-frame-pointer'
+    ],
+    CXXFLAGS=[
+        '-fpermissive', '-fno-rtti', '-fno-exceptions', '-fno-use-cxa-atexit',
+    ],
+    CPPDEFINES={
+        'REPO_URL': '\\"$REPO_URL\\"',
```

Comment:
We might want to use `shlex.quote()` for these. Just a suggestion, take it or leave it.

Reply:
Ooh yeah, good idea. I already had some interesting ideas for passing through these build variables as defines.

Reply:
I tried using `shlex` and wasn't able to get it to work. I think it might have something to do with the fact that we're trying to access variables that aren't yet "commited" to the environment.

Reply:
Hmm yeah, I guess I'll have to play with it a bit. I think it might be possible with a little bit of creativity. Thinking about it a little more I don't think it's really necessary since there shouldn't be any adversarial input to the build system, but I'm not opposed to using it if we can get it to work.